hazelcast-client/src/main/java/com/hazelcast/client/proxy/ClientConditionProxy.java

```diff
@@ -91,8 +91,11 @@ private boolean doAwait(long time, TimeUnit unit, long threadId) throws Interrup
     @Override
     public boolean awaitUntil(Date deadline) throws InterruptedException {
         long until = deadline.getTime();
-        final long timeToDeadline = until - Clock.currentTimeMillis();
-        return await(timeToDeadline, TimeUnit.MILLISECONDS);
+        final long durationMs = until - Clock.currentTimeMillis();
+        if (durationMs <= 0) {
```

Comment:
What about the other methods taking "long time" as a parameter (e.g. public boolean await(long time, TimeUnit unit))? Is it better to do this check at the method taking long time as a parameter?